### PR TITLE
utxo: Add some tests for real-time time locks

### PR DESCRIPTION
**Pull request by iljakuklic**
These tests use `proptest` for randomized property-based testing.

By default, the `prop_*` functions are run with 256 different inputs.

### CODE REVIEW

File: pallets/utxo/src/tests.rs [hunk 1]
```diff
@@ -601,3 +602,96 @@ fn test_send_to_address() {
         );
     })
 }
+
+// Generate block time in seconds
+fn gen_block_time_real() -> impl Strategy<Value = RawBlockTime> {
+    (RawBlockTime::THRESHOLD..RawBlockTime::THRESHOLD + 1_000_000_000).prop_map(RawBlockTime::new)
```
Comment (muursh) [inline pallets/utxo/src/tests.rs:608]: What is this 1_000_000_000?

**Review by muursh — APPROVED** [review]
Looks good to me. Just not sure where the 1bn comes from

File: pallets/utxo/src/tests.rs (hunk 1 above)
Comment (iljakuklic) [reply pallets/utxo/src/tests.rs:608]: That is just an arbitrary upper bound for generated timestamps. Can change it to `2 * THRESHOLD`, that may be more clear.

Comment (iljakuklic) [reply pallets/utxo/src/tests.rs:608]: Just an arbitrary upper bound fr generated timestamps. The result is `3 * THRESHOLD` so can change it to that for more clarity.

Comment (muursh) [reply pallets/utxo/src/tests.rs:608]: Either that or just a comment. I'm fine with either.

File: pallets/utxo/src/script.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ impl RawBlockTime {
     // Values less than this constant are considered to be a block number, other values are
     // interpreted as UNIX time stamp. This constant is identical to what is used in Bitcoin.
     // At 1min/block, the block numbers run out in about 900 years after the genesis block.
-    const THRESHOLD: u64 = 500_000_000u64;
+    pub const THRESHOLD: u64 = 500_000_000u64;
```
Comment (TheQuantumPhysicist) [inline pallets/utxo/src/script.rs:35]: I think you don't have to make this public if you make the tests under the `#[tests]`. This is a huge problem in C++, that you cannot access private variables in tests. But I think rust solved that problem. Please double check.

Comment (iljakuklic) [reply pallets/utxo/src/script.rs:35]: I was thinking of something like:

```rust
#[cfg(test)]
const THRESHOLD: u64 = 500_000_000u64;
#[cfg(test)]
pub const THRESHOLD: u64 = 500_000_000u64;
```

But found it a bit too verbose for a small issue like this. If Rust has a better mechanism, that's great.

Comment (iljakuklic) [reply pallets/utxo/src/script.rs]: So in the end, I moved the test helper function that uses the THRESHOLD constant into the `test` submodule of this module so it has access to struct fields without making them public. The function is then imported in `test.rs`.

**Review by TheQuantumPhysicist — APPROVED** [review]

**Comment by iljakuklic** [comment]
Added two more tests, one checking time locks for `(now - 1, now, now + 1)`, other to check overflow for timestamps close to `u64::MAX`.

**Comment by muursh** [comment]
New changes look good to me too

**Review by TheQuantumPhysicist — APPROVED** [review]
Looks good to me!
